### PR TITLE
[new opmath] `tests/optimize/`

### PR DESCRIPTION
**Pull request by Qottmann**
`Sum.sparse_matrix()` uses the pauli rep to compute the sparse matrix, which has a different default wire order which causes a failure. The update is in a subfunction of `RiemannianGradientOptimizer` to use explicit wire_order argument of the tape.

- [x] `tests/optimize` pass except for one mentioned below
- [x] `tests/optimize/test_qng.py::TestOptimize::test_step_and_cost_autograd_with_gen_hamiltonian` deferred to [this story](https://app.shortcut.com/xanaduai/story/58893/update-tests-tests-optimize-test-qng-py-testoptimize-test-step-and-cost-autograd-with-gen-hamiltonian)

builds on top of https://github.com/PennyLaneAI/pennylane/pull/5362/

[sc-58399]

### CODE REVIEW

File: tests/optimize/test_spsa.py [hunk 1]
```diff
@@ -442,6 +442,44 @@ def cost(params):
         ):
             opt.step(cost, params)
 
+    @pytest.mark.usefixtures("use_legacy_opmath")
+    @pytest.mark.slow
+    def test_lighting_device(self):
```
Comment (astralcai) [inline tests/optimize/test_spsa.py:447]: I thought legacy test cases should be post-fixed with `_legacy`?

Comment (Qottmann) [reply tests/optimize/test_spsa.py:447]: Indeed, will change 👍 `_legacy_opmath()` in fact

Comment (Qottmann) [reply tests/optimize/test_spsa.py:447]:
```suggestion
    def test_lighting_device_legacy_opmath(self):
```

File: tests/optimize/test_qng.py [hunk 2]
```diff
@@ -73,6 +73,7 @@ def circuit(params):
         assert np.allclose(step1, expected_step)
         assert np.allclose(step2, expected_step)
 
+    @pytest.mark.usefixtures("use_legacy_opmath")  # currently fails with new opmath
     def test_step_and_cost_autograd_with_gen_hamiltonian(self):
```
Comment (astralcai) [inline tests/optimize/test_qng.py:77]: I believe the strategy for this is to have two test cases, and xfail the failing one?

Comment (Qottmann) [reply tests/optimize/test_qng.py:77]: Yep I created a story for that https://app.shortcut.com/xanaduai/story/58893/update-tests-tests-optimize-test-qng-py-testoptimize-test-step-and-cost-autograd-with-gen-hamiltonian

Comment (Qottmann) [reply tests/optimize/test_qng.py:77]: Oh but yes you are right I didnt at an xfailing test, only for the lightning one

**Review by astralcai — APPROVED** [review]
🎸

**Review by PietropaoloFrisoni — APPROVED** [review]
Thanks!

**Pull request merged**